ifrs17/Import/Importers.ipynb

```diff
@@ -503,18 +503,32 @@
+                "\n        case ImportFormats.Cashflow:",
+                "\n        case ImportFormats.Actual: ",
+                "\n        case ImportFormats.Opening:",
+                "\n        {",
+                "\n            if(args.Scenario != null) return args.RepeatOnce().ToArray();",
+                "\n            var secondaryArgs = await dataSource.Query<PartitionByReportingNodeAndPeriod>() ",
+                "\n                .Where(x => x.ReportingNode == args.ReportingNode && x.Year == args.Year && x.Month == args.Month && x.Scenario != null)",
+                "\n                .Select(x => new ImportArgs(x.ReportingNode, x.Year, x.Month, default(Periodicity), x.Scenario, format)).ToArrayAsync();",
+                "\n   ",
+                "\n            if(secondaryArgs.Any()) ApplicationMessage.Log(Warning.ScenarioReCalculations, String.Join(\", \", secondaryArgs.Select(x => x.Scenario)));",
+                "\n            allArgs = args.RepeatOnce().Concat(secondaryArgs).ToArray();",
+                "\n            break;",
+                "\n        }",
+                "\n        default: ",
                 "\n        {",
                 "\n            if(args.Scenario == null) {",
-                "\n                var scenariosWithYieldCurves = await dataSource.Query<YieldCurve>().Where(x => x.Year == args.Year && x.Month == args.Month && x.Scenario != null)",
-                "\n                    .Select(x => x.Scenario).Distinct().ToArrayAsync();",
+                "\n                var scenarios = (await dataSource.Query<YieldCurve>().Where(x => x.Year == args.Year && x.Month == args.Month && x.Scenario != null).Select(x => x.Scenario).ToArrayAsync())",
+                "\n                         .Concat(await dataSource.Query<DataNodeParameter>().Where(x => x.Year == args.Year && x.Month == args.Month && x.Scenario != null).Select(x => x.Scenario).ToArrayAsync())",
+                "\n                         .Distinct();",
                 "\n                var targetPartitions = await dataSource.Query<PartitionByReportingNodeAndPeriod>()",
-                "\n                    .Where(x => x.Year == args.Year && x.Month == args.Month && !scenariosWithYieldCurves.Contains(x.Scenario)).OrderBy(x => x.Scenario).ToArrayAsync();",
+                "\n                    .Where(x => x.Year == args.Year && x.Month == args.Month && !scenarios.Contains(x.Scenario)).OrderBy(x => x.Scenario).ToArrayAsync();",
                 "\n                var targetScenarios = targetPartitions.Where(x => x.Scenario != null).Select(x => x.Scenario);",
                 "\n                if(targetScenarios.Any()) ApplicationMessage.Log(Warning.ScenarioReCalculations, String.Join(\", \", targetScenarios));",
                 "\n                allArgs = targetPartitions.Select(x => new ImportArgs(x.ReportingNode, x.Year, x.Month, default(Periodicity), x.Scenario, ImportFormats.Cashflow)).ToArray();",
```

Comment:
the format is set to Cashflow here. While for YC I see as necessary, for a generic ParameterUpdated it might not be necessary. Given the parameters we have now it would be enough format Actual. This would speed up import. On the other side the code would get more complex having to decide which format is the best given the scenario. Also, we might add new parameters in the future making the logic even more complex. 
In summary, I think it is a good decision to keep Cashflow format here.

Reply:
Yes, thanks! I also think it is quite a general rule that would remain valid for more parameters.